### ISSUE TITLE
Dependency resolution errors installing into new NuxtJS app

### ISSUE THREAD

**Issue by DaveMuirhead** [nuxt-community/apollo-module#379]
### Version

[v4.0.1-rc.1](https://github.com/apollo-module/releases/tag/v4.0.1-rc.1)

### Reproduction link

[https://discord.com/channels/473401852243869706/507590246070878210/805808885390245908](https://discord.com/channels/473401852243869706/507590246070878210/805808885390245908)

### Steps to reproduce

npx create-nuxt-app nuxt-graphql
cd nuxt-graphql
npm install @nuxtjs/apollo

### What is expected ?

Installation of @nuxtjs/apollo into NuxtJS app

### What is actually happening?

Dependency resolution errors. I can provide the complete NPM debug log if you want it. This error is also reported by other users today on the #apollo channel of the Nuxt.js Discord server.

### Additional comments?

[594][davem: nuxt-graphql]$ npm install --save @nuxtjs/apollo
npm WARN ERESOLVE overriding peer dependency
npm WARN Found: graphql@15.5.0
npm WARN node_modules/graphql
npm WARN   peer graphql@"^0.11.0 || ^0.12.0 || ^0.13.0 || ^14.0.0 || ^15.0.0" from apollo-cache-inmemory@1.6.6
npm WARN   node_modules/apollo-cache-inmemory
npm WARN     peer apollo-cache-inmemory@"^1.6.6" from @nuxtjs/apollo@4.0.1-rc.5
npm WARN     node_modules/@nuxtjs/apollo
npm WARN     1 more (vue-cli-plugin-apollo)
npm WARN   8 more (apollo-client, apollo-link, graphql-tag, ...)
npm WARN
npm WARN Could not resolve dependency:
npm WARN peer graphql@"^0.11.0 || ^0.12.0 || ^0.13.0 || ^14.0.0" from apollo-link-persisted-queries@0.2.2
npm WARN node_modules/apollo-link-persisted-queries
npm WARN   apollo-link-persisted-queries@"^0.2.2" from vue-cli-plugin-apollo@0.22.2
npm WARN   node_modules/vue-cli-plugin-apollo
npm WARN ERESOLVE overriding peer dependency
npm WARN Found: graphql@15.5.0
npm WARN node_modules/graphql
npm WARN   peer graphql@"^0.11.0 || ^0.12.0 || ^0.13.0 || ^14.0.0 || ^15.0.0" from apollo-cache-inmemory@1.6.6
npm WARN   node_modules/apollo-cache-inmemory
npm WARN     peer apollo-cache-inmemory@"^1.6.6" from @nuxtjs/apollo@4.0.1-rc.5
npm WARN     node_modules/@nuxtjs/apollo
npm WARN     1 more (vue-cli-plugin-apollo)
npm WARN   8 more (apollo-client, apollo-link, graphql-tag, ...)
npm WARN
npm WARN Could not resolve dependency:
npm WARN peer graphql@"^0.10.5 || ^0.11.3 || ^0.12.0 || ^0.13.0 || ^14.0.0" from graphql-subscriptions@1.1.0
npm WARN node_modules/graphql-subscriptions
npm WARN   graphql-subscriptions@"^1.1.0" from vue-cli-plugin-apollo@0.22.2
npm WARN   node_modules/vue-cli-plugin-apollo
npm WARN   1 more (apollo-server-express)
npm WARN ERESOLVE overriding peer dependency
npm WARN Found: graphql@15.5.0
npm WARN node_modules/graphql
npm WARN   peer graphql@"^0.11.0 || ^0.12.0 || ^0.13.0 || ^14.0.0 || ^15.0.0" from apollo-cache-inmemory@1.6.6
npm WARN   node_modules/apollo-cache-inmemory
npm WARN     peer apollo-cache-inmemory@"^1.6.6" from @nuxtjs/apollo@4.0.1-rc.5
npm WARN     node_modules/@nuxtjs/apollo
npm WARN     1 more (vue-cli-plugin-apollo)
npm WARN   8 more (apollo-client, apollo-link, graphql-tag, ...)
npm WARN
npm WARN Could not resolve dependency:
npm WARN peer graphql@"^0.10.0 || ^0.11.0 || ^0.12.0 || ^0.13.0 || ^14.0.0" from @apollographql/graphql-language-service-interface@2.0.2
npm WARN node_modules/@apollographql/graphql-language-service-interface
npm WARN   @apollographql/graphql-language-service-interface@"^2.0.2" from apollo-language-server@1.25.0
npm WARN   node_modules/apollo-language-server
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! Found: graphql@15.5.0
npm ERR! node_modules/graphql
npm ERR!   peer graphql@"^0.11.0 || ^0.12.0 || ^0.13.0 || ^14.0.0 || ^15.0.0" from apollo-cache-inmemory@1.6.6
npm ERR!   node_modules/apollo-cache-inmemory
npm ERR!     peer apollo-cache-inmemory@"^1.6.6" from @nuxtjs/apollo@4.0.1-rc.5
npm ERR!     node_modules/@nuxtjs/apollo
npm ERR!       @nuxtjs/apollo@"*" from the root project
npm ERR!     apollo-cache-inmemory@"^1.6.6" from vue-cli-plugin-apollo@0.22.2
npm ERR!     node_modules/vue-cli-plugin-apollo
npm ERR!       vue-cli-plugin-apollo@"^0.22.2" from @nuxtjs/apollo@4.0.1-rc.5
npm ERR!       node_modules/@nuxtjs/apollo
npm ERR!         @nuxtjs/apollo@"*" from the root project
npm ERR!   peer graphql@"^0.11.0 || ^0.12.0 || ^0.13.0 || ^14.0.0 || ^15.0.0" from apollo-client@2.6.10
npm ERR!   node_modules/apollo-client
npm ERR!     peer apollo-client@"^2.0.0" from vue-apollo@3.0.5
npm ERR!     node_modules/vue-apollo
npm ERR!       vue-apollo@"^3.0.5" from @nuxtjs/apollo@4.0.1-rc.5
npm ERR!       node_modules/@nuxtjs/apollo
npm ERR!         @nuxtjs/apollo@"*" from the root project
npm ERR!     apollo-client@"^2.6.10" from vue-cli-plugin-apollo@0.22.2
npm ERR!     node_modules/vue-cli-plugin-apollo
npm ERR!       vue-cli-plugin-apollo@"^0.22.2" from @nuxtjs/apollo@4.0.1-rc.5
npm ERR!       node_modules/@nuxtjs/apollo
npm ERR!         @nuxtjs/apollo@"*" from the root project
npm ERR!   7 more (apollo-link, graphql-tag, vue-cli-plugin-apollo, ...)
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer graphql@"^0.10.0 || ^0.11.0 || ^0.12.0 || ^0.13.0 || ^14.0.0" from @apollographql/graphql-language-service-parser@2.0.2
npm ERR! node_modules/@apollographql/graphql-language-service-interface/node_modules/@apollographql/graphql-language-service-parser
npm ERR!   @apollographql/graphql-language-service-parser@"^2.0.0" from @apollographql/graphql-language-service-interface@2.0.2
npm ERR!   node_modules/@apollographql/graphql-language-service-interface
npm ERR!     @apollographql/graphql-language-service-interface@"^2.0.2" from apollo-language-server@1.25.0
npm ERR!     node_modules/apollo-language-server
npm ERR!       apollo-language-server@"^1.25.0" from apollo@2.32.1
npm ERR!       node_modules/apollo
npm ERR!       1 more (apollo-codegen-core)
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR!
npm ERR! See /Users/davem/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/davem/.npm/_logs/2021-02-01T14_45_22_425Z-debug.log

<div align="right"><sub><em>This bug report is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/apollo-module/issues/c359">#c359</a>)</em></sub></div>

**Comment by Akkarine** [comment]
I blame https://github.com/Akryum/vue-cli-plugin-apollo for using incompatible and deprecated libraries (also `@hapi/...`). Looks like this library requires some [maintenance](https://github.com/Akryum/vue-cli-plugin-apollo/issues/523).

Fixed `graphql` @14.7.0
![image](https://user-images.githubusercontent.com/13115270/110453894-ba7cd800-80d7-11eb-8d9c-8c9224432634.png)  

Fixed `graphql` @15.5.0:
![image](https://user-images.githubusercontent.com/13115270/110453900-bcdf3200-80d7-11eb-90d6-0745354f6d52.png)  

```log
npm WARN ERESOLVE overriding peer dependency
npm WARN Found: graphql@15.5.0
npm WARN node_modules/graphql
npm WARN   peer graphql@"^14.7.0 || ^15.0.0" from the root project
npm WARN   15 more (apollo-cache-inmemory, apollo-client, apollo-link, ...)
npm WARN
npm WARN Could not resolve dependency:
npm WARN peer graphql@"^0.11.0 || ^0.12.0 || ^0.13.0 || ^14.0.0" from apollo-link-persisted-queries@0.2.2
npm WARN node_modules/apollo-link-persisted-queries
npm WARN   apollo-link-persisted-queries@"^0.2.2" from vue-cli-plugin-apollo@0.22.2
npm WARN   node_modules/vue-cli-plugin-apollo
npm WARN deprecated @hapi/topo@3.1.6: This version has been deprecated and is no longer supported or maintained
npm WARN deprecated @hapi/bourne@1.3.2: This version has been deprecated and is no longer supported or maintained
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
npm WARN deprecated har-validator@5.1.5: this library is no longer supported
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
npm WARN deprecated chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
npm WARN deprecated fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
npm WARN deprecated chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
npm WARN deprecated fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
npm WARN deprecated chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
npm WARN deprecated @hapi/address@2.1.4: Moved to 'npm install @sideway/address'
npm WARN deprecated mkdirp@0.3.0: Legacy versions of mkdirp are no longer supported. Please update to mkdirp 1.x. (Note that the API surface has changed to use Promises in 1.x.)
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
npm WARN deprecated @hapi/hoek@8.5.1: This version has been deprecated and is no longer supported or maintained
npm WARN deprecated @hapi/joi@15.1.1: Switch to 'npm install joi'
npm WARN deprecated core-js@2.6.12: core-js@<3 is no longer maintained and not recommended for usage due to the number of issues. Please, upgrade your dependencies to the actual version of core-js@3.
```

**Comment by DaveMuirhead** [comment]
Tried again just now and `npm install @nuxtjs/apollo` appeared to work. I assume, therefore, that this was some intermittent issue that is now resolved.
